Sources/DataPipeline/CIODataPipelineEvents.swift

```diff
@@ -0,0 +1,116 @@
+import CioInternalCommon
+import Foundation
+
+// MARK: - Typed Event Signatures
+
+public extension CustomerIO {
```

Comment:
I think it would be a good idea to create an [APITest](https://github.com/customerio/customerio-ios/blob/main/Tests/Tracking/APITest.swift) for this new module. It has been really helpful to make sure that the public-facing API compiles for customers as intended. 

If you agree, could you make sure this is added as a task to a CDP ticket to do this? It's OK if this is not part of this PR if that's out of scope.

Reply:
Agreed, I was thinking it could be added to Ready for Production tickets where we can add even more test cases related.

Reply:
I am OK with that. 

If that's the case, could you add a // TODO or // FIXME comment to this file specifying that we are planning on that?